lib/test_formulae.rb

```diff
@@ -15,22 +15,22 @@ def initialize(tap:, git:, dry_run:, fail_fast:, verbose:)
 
       protected
 
+      def cached_event_json
+        return unless (event_json = artifact_cache/"event.json").exist?
+
+        event_json
+      end
+
       def previous_github_sha
         return if tap.blank?
         return unless repository.directory?
         return if ENV["GITHUB_ACTIONS"].blank?
 
-        @previous_sha ||= begin
-          event_path = ENV.fetch("GITHUB_EVENT_PATH")
-          event_payload = JSON.parse(File.read(event_path))
-
-          before = event_payload.fetch("before", "")
-          test git, "-C", repository, "fetch", "origin", before if before.present?
-
-          before
-        end
+        event_path = cached_event_json
+        event_path ||= Pathname(ENV.fetch("GITHUB_EVENT_PATH"))
+        event_payload = JSON.parse(event_path.read)
 
-        @previous_sha
+        event_payload.fetch("before", nil)
```

Comment:
So if we have the current event we take the previous (before) sha, but if we download a previous artifact don't we want the current sha from that payload? Or will that payload be the current run from an earlier job?